move docs to karmada website (#2323)
karmada-io/karmada: docs/README.md

> All documentations for the Karmada is hosted at [karmada-io/website](https://github.com/karmada-io/website).

Reviewer: ```suggestion
Karmada documentation is all hosted on [karmada-io/website](https://github.com/karmada-io/website).
```

Author: Done.
